### ISSUE TITLE
private method not used 

### ISSUE THREAD

**Issue by IlyasYOY** [pmd/pmd#4607]
**Affects PMD Version:** 7.0.0-rc3

**Rule:** Best Practices/UnusedPrivateMethod
https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:** false positive when private method only used as method reference.

**Code Sample demonstrating the issue:**

```java
    public Y do() {
        return provider.find().map(this::usedButUnused).get();
    }

    private Y usedButUnused(X x) {
        // ...
    }
```

**Expected outcome:** method should not be marked as unused.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
I cannot reproduce this. I used the following MWE:
```java
import java.util.Optional;
interface A {}
interface B {}
public class Foo {
    B foo(Optional<A> opt) {
        return opt.map(this::aToB).get();
    }

    private B aToB(A a) { return null; }

}
```

I suspect some types in your auxclasspath are missing (for instance X, Y, the type of `provider`, the type of `this`). All these classes must be on the `--aux-classpath` for PMD to do semantic analysis properly. Run with `--debug` and look for logs of the form `Cannot resolve ...` to find out if some classes are missing.

This could be a legitimate issue if your `usedButUnused` has overloads for instance. However for the reported case, and if PMD is properly configured, there is no issue. I'm closing the issue but feel free to comment further if you need help or still have the issue.
